KDECompilerSettings (of extra-cmake-modules 5.40) sets "-std=iso9899:1990" that is
too restrictive for C build with Qt *.h files: comments in c++ style. 
Let's use C99 instead.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,8 @@
 include(KDEInstallDirs)
 include(KDECMakeSettings)
 include(KDECompilerSettings NO_POLICY_SCOPE) # needs to be first, as set policies influence following macros
+string(REPLACE " -std=iso9899:1990" "" CMAKE_C_FLAGS " ${CMAKE_C_FLAGS} ")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
 
 include(ECMOptionalAddSubdirectory)
 include(ECMInstallIcons)
